noir/noir-repo/acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -44,11 +44,15 @@ pub(crate) fn evaluate_binary_field_op<F: AcirField>(
         BinaryFieldOp::Mul => MemoryValue::new_field(a * b),
         BinaryFieldOp::Div => MemoryValue::new_field(a / b),
         BinaryFieldOp::IntegerDiv => {
-            let a_big = BigUint::from_bytes_be(&a.to_be_bytes());
-            let b_big = BigUint::from_bytes_be(&b.to_be_bytes());
+            if b.is_zero() {
```

Comment:
do we want to allow division by zero?
I think this behaviour does not match Noir behaviour.

Reply:
It was already returning zero for divisions by zero in the integer version. I think without this directive_quotient fails in some cases. We could add branching in directive_quotient, but we should decide on wether to fail for division by zero or not in general

Reply:
I've pushed a change that errors on division by zero on fields and integers, lets see if it passes all the testing suites

Reply:
Yep seems to be fine!